### ISSUE TITLE
Consider publishing pulumi to winget packages repository

### ISSUE THREAD

**Issue by rodrigoscna** [pulumi/pulumi#4676]
It would be nice to have pulumi installed by just running `winget install pulumi`.

**Comment by jaxxstorm** [comment]
I had a look at this because I thought it might be interesting as well!

Adding a package seems relatively straightforward, you just need to add a version config to the packages repo:
https://github.com/microsoft/winget-pkgs/tree/master/manifests

However, what's currently not clear is what the schema definition for those packages is. For example, [here's](https://docs.microsoft.com/en-us/windows/package-manager/package/manifest?tabs=minschema%2Ccompschema#complete-schema) the complete YAML schema, however it seems that the assumption here is that the winget package _executes_ an installer, rather than just copy a binary

**Comment by rodrigoscna** [comment]
Yeah, it's basically an auto-installer which downloads the setup file and runs it.

I believe you would need to add to the CI a pull request to their repository for every new version released that would have the URL for the newest installer.

**Comment by adriantorrie** [comment]
This would be good to have

**Comment by adriantorrie** [comment]
`winget` released at v1.0 now:
https://devblogs.microsoft.com/commandline/windows-package-manager-1-0/

Makes `winget` available for us in a corporate environment to use as a supported CLI now. Having pulumi as an available 
 `winget` package would be great.

**Comment by Zaid-Ajaj** [comment]
Currently blocked by [winget-cli: Support installing .zip files #140](https://github.com/microsoft/winget-cli/issues/140)

**Comment by Zaid-Ajaj** [comment]
OK, this _was_ blocked because `winget` doesn't know how to install zip files which is what pulumi publishes to the releases. However, I was able to programmatically generate an [MSI installer](https://github.com/pulumi/pulumi-winget/releases/tag/v3.26.1) from the corresponding zip files (retrieved from pulumi releases). 

After releasing the MSI, a winget manifest is generated from it and sent as an [automated PR to the winget](https://github.com/microsoft/winget-pkgs/pull/54644) registry

All that is left is for the winget registry to accept the PR to make it available from `winget install pulumi`.

**Comment by jedieaston** [comment]
Thanks for setting this up!

Looks like @pulumi-bot needs to sign the Microsoft CLA for the PR to pass CI. [Once that's fixed](https://github.com/microsoft/winget-pkgs/pull/54644#issuecomment-1069781619), this should be good to go.

**Comment by Zaid-Ajaj** [comment]
@rodrigoscna, @adriantorrie Pulumi is now available through `winget` as of version v3.26.1
```
winget install pulumi
```
Will install the latest version of pulumi and subsequent versions will be made available as soon as they are released. 

Please give it a try and let us know if you encounter any issues. In case of any, please consider reopening this issue. 

Thanks @jedieaston for reviewing the PR for adding pulumi to the winget registry 🙏

**Comment by vedantmgoyal2009** [comment]
Hello,

Can you please update the automation to use a multi-manifest structure instead of a singleton manifest? I'm saying this because they will be soon deprecated by Windows Package Manager.

**Comment by jedieaston** [comment]
> they will be soon deprecated by Windows Package Manager.

soon, as in 1-2 years, maybe, if someone can find a good reason to remove them. It's a breaking change to the format, and hasn't even been seriously discussed (that I've seen) outside of a couple Gitter messages.

(so don't worry about it).
